devicetypes/smartthings/ezex-smart-electric-switch.src/ezex-smart-electric-switch.groovy

```diff
@@ -0,0 +1,124 @@
+        capability "Power Meter"
+        capability "Actuator"
+        capability "Switch"
+        capability "Refresh"
+        capability "Health Check"
+        capability "Sensor"
+        capability "Configuration"
+
+        fingerprint profileId: "0104", deviceId:"0053", inClusters: "0000, 0003, 0004, 0006, 0B04, 0702", outClusters: "0019", manufacturer: "", model: "E240-KR116Z-HA", deviceJoinName: "Smart Electric Switch"
+    }
+
+    tiles(scale: 2){
+        multiAttributeTile(name:"switch", type: "generic", width: 6, height: 4, canChangeIcon: true){
+                tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+                        attributeState("on", label: '${name}', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#00a0dc")
+                        attributeState("off", label: '${name}', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff")
+                }
+        }
+        valueTile("power", "device.power", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} W'
+        }
+        valueTile("energy", "device.energy", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} kWh'
+        }
+        standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'reset kWh', action:"reset"
+        }
+
+        main(["switch"])
+        details(["switch","power","energy","refresh","reset"])
+    }
+}
+
+def getATTRIBUTE_READING_INFO_SET() { 0x0000 }
+def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }
+
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        log.info event
+        sendEvent(event)
+    } else {
+        List result = []
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "Desc Map: $descMap"
+        
+        List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
+        descMap.additionalAttrs.each {
+            attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+        }
+        attrData.each {
+                def map = [:]
+                if (it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_HISTORICAL_CONSUMPTION) {
+                        log.debug "power"
+                        map.name = "power"
+                        map.value = zigbee.convertHexToInt(it.value)/1000
+                }
+                if (it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_READING_INFO_SET) {
```

Comment:
More of nit-picking, but don't forget the `else if` instead of another `if` here. In the scheme of things it probably doesn't matter, and the Groovy VM likely optimizes it, but it should help cut down a few CPU cycles which should ease the load on the cloud servers.

Reply:
ok，I‘ll add it.